src/main/java/eu/enmeshed/model/relationships/RelationshipAuditLog.java

```diff
@@ -0,0 +1,46 @@
+package eu.enmeshed.model.relationships;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.ZonedDateTime;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+@Builder
+public class RelationshipAuditLog {
+    private ZonedDateTime createdAt;
+    private String createdBy;
+    private String createdByDevice;
+    private Reason reason;
+    private RelationshipStatus oldStatus;
+    private RelationshipStatus newStatus;
+
+    public enum Reason {
```

Comment:
I don't think so. As this enum is inside the other class it will be called as `RelationshipAuditLogEntry.Reason`. `RelationshipAuditLogEntry.RelationshipAuditLogEntryReason` would be overkill.